tests/unit/select_connection_ioloop_tests.py

```diff
@@ -35,10 +46,13 @@ def setUp(self):
         self.ioloop = select_connection.IOLoop()
 
     def tearDown(self):
-        self.ioloop.remove_timeout(self.fail_timer)
+        if hasattr(self, 'fail_timer'): # some test don't use this timer.
+            self.ioloop.remove_timeout(self.fail_timer)
         self.ioloop = None
 
     def start(self):
+        ''' Setup timeout handler for detecting 'no-activity'
+            and start polling. '''
         self.fail_timer = self.ioloop.add_timeout(self.TIMEOUT, self.on_timeout)
```

Comment:
Instead of the messy remove_timeout logic in `tearDown`, do this instead:

```
        fail_timer = self.ioloop.add_timeout(self.TIMEOUT, self.on_timeout)
        self.addCleanup(self.ioloop.remove_timeout, fail_timer)
```

fail_timer can no be a local variable and there is no need to create the instance member `self.fail_timer`